plugins/vpc-bridge/config/k8s/pipe.go

```diff
@@ -0,0 +1,197 @@
+	"net"
+	"time"
+
+	winio "github.com/Microsoft/go-winio"
+	log "github.com/cihub/seelog"
+	"golang.org/x/sys/windows"
+)
+
+const (
+	//PipeConnectionTimeout Represents timeout for reading from connection object.
+	PipeConnectionTimeout = 1 * time.Minute
+
+	//NamedPipePathPrefix Represents prefix for named pipe path.
+	NamedPipePathPrefix = `\\.\pipe\aws-vpc-k8s-connector-%s`
+)
+
+// PipeReadResult Represents output received from named pipe.
+type PipeReadResult struct {
+	stdout string // Represents message with output data.
+	stderr error  // Represents error while getting output.
+}
+
+// PipeOperator Represents interface with all operations on named piped.
+type PipeOperator interface {
+	Start(ctx context.Context) error               // Start listener and accept connections on named pipe.
+	GetOutput(ctx context.Context) (string, error) // Get output from named pipe.
+	GetPath() string                               // Get named pipe path.
+	Close() error                                  // Close listener.
+}
+
+// Pipe Represents pipe.
+type Pipe struct {
+	Path       string              // Represents named pipe path.
+	listener   net.Listener        // Represents listener on named pipe.
+	resultChan chan PipeReadResult // Represents results chan to receive output from go routine.
+}
+
+// NewPipe Create Pipe object with random named pipe path.
+func NewPipe() (PipeOperator, error) {
+	// Generate GUID random string.
+	g, err := windows.GenerateGUID()
+	if err != nil {
+		return nil, fmt.Errorf("error creating unique name for named pipe: %w", err)
+	}
+	// Create pipe path with prefix.
+	pipeName := fmt.Sprintf(NamedPipePathPrefix, g.String())
+	return &Pipe{Path: pipeName}, nil
+}
+
+// Start listener on named pipe. Start accepting connections on named pipe in separate go routine.
+// Creates result chan to get output from separate go routine.
+func (p *Pipe) Start(ctx context.Context) error {
+
+	// Dials and start listener on named pipe.
+	listener, err := p.listen()
+	if err != nil {
+		return fmt.Errorf("error creating listener: %w", err)
+	}
+	p.setListener(listener) // Set listener on Pipe object.
+
+	// Creating result channel to get output from pipe.
+	resultChan := make(chan PipeReadResult)
+	p.setResultChan(resultChan) // Set results chan on Pipe object
+
+	// Accept connections and read pipe in go routine as these operations are blocking.
+	go p.acceptConnections(listener, resultChan)
+
+	return nil
+}
+
+// GetPath get name/path for named pipe.
+func (p *Pipe) GetPath() string {
+	return p.Path
+}
+
+// GetOutput Get output received from named pipe using results chan.
+func (p *Pipe) GetOutput(ctx context.Context) (string, error) {
+
+	// Get result chan from Pipe object.
+	resultChan := p.getResultChan()
+
+	// Get output from named pipe using result chan.
+	var result PipeReadResult
+	select {
+	case result = <-resultChan:
+	case <-ctx.Done():
+		return "", fmt.Errorf("timeout getting output from named pipe result chan")
```

Comment:
You can wrap it but you should return ctx.Err() here to know the real error. Timeout is an assumption here

Reply:
done.